### DISCUSSION TITLE
Query Mongo/Redis directly from page inside getServerSideProps

### DISCUSSION THREAD

**Discussion by karkovich** [vercel/next.js#13734] [Help] [answered]
If I want to query with mongoose/Redis directly inside `getServerSideProps`

```javascript
export const getServerSideProps = (context) => {
  // don't want to connect here!
  const doc = await  Model.findOne(....) // or mayby redis query
  return {
    props: {...}
  }
}
```

where should I connect to mongo/redis? I don't want to create connection every page.
I want to run the `next.js` server (`npm start`) and create the connection before the client visiting. so, each page could use the same connection?

**Comment by lfades** [comment, marked as answer]
You can connect there but make sure that every time you try to connect, return a connection that was already created, you can see how here: [examples/with-mongodb-mongoose/pages/index.js](https://github.com/vercel/next.js/blob/canary/examples/with-mongodb-mongoose/pages/index.js)
